snapshot-labs/snapshot-sidekick
# fix: avoid duplicate processing of long running jobs

## 🧿 Current issues / What's wrong ?

The current queue system has been put in place to avoid queuing the same job multiple time. It is not preventing the same job to be processed multiple times, when the processing is taking very long, because it was assumed that job processing finish before next queue polling (every 15s)

## 💊 Fixes / Solution

Add a check to prevent the same job to be picked up again by the queue, when it is already being processed.

## 🚧 Changes

- Add a check to avoid processing the same job multiple time

## 🛠️ Tests

- Run `yarn dev`
- Run ` curl -X POST http://localhost:3005/api/votes/0x6b703b90d3cd1f82f7c176fc2e566a2bb79e8eb6618a568b52a4f29cb2f8d57b` (this is a very big file, and taking more than 15s to process, ideal for testing)
- Monitor the `yarn dev` console output
- It should pick and process the job the first time
- After 15s, the next polling should still detect that job in the queue, but skip it, with the output `[queue] Skip: 0x6b703b90d3cd1f82f7c176fc2e566a2bb79e8eb6618a568b52a4f29cb2f8d57b currently being processed`
